RFC/src/RFC-0300_DAN.md

```diff
@@ -1 +1,204 @@
+
+#### Execution of instructions
+VNs are expected to manage the state of digital assets on behalf of digital asset issuers. They receive fees as reward
+for doing this.
+* Digital assets consist of an initial state plus a set of state transition rules. These rules are set by the Tari
+  protocol, but will usually provide parameters that must be specified by the [Asset Issuer].
+* The set of VNs that participate in managing state of a specific DA is called a [Committee]. A committee is selected during the asset
+issuance process and membership of the committee can be updated at [Checkpoint]s.
+* It is the VNs responsibility to ensure that every state change in a digital asset conforms to the contract's rules.
+* VNs accept digital asset [Instructions] from clients and peers. [Instructions] allow for creating, updating, and
+  expiring digital assets on the DAN.
+* VNs provide additional collateral when accepting an offer to manage an asset, which is stored in a multi-signature
+  contract on the base layer. This collateral can be taken from the VN if it is proven that the VN engaged in
+  malicious behaviour.
+* VNs participate in fraud proof validations in the event of consensus disputes (which could result in the malicious VN's
+  collateral being slashed).
+* Digital asset metadata (e.g. large images) are managed by VNs, but whether the data is considered part of the state
+  (and thus checkpointed) or out of state depends on the type of digital asset contract employed.
+
+#### Fees
+Fees will be paid to VNs based on the amount of work they did during a checkpoint period. The fees will be paid from a fee pool which will be collected
+and reside in a UTXO that is accessible by the committee. The exact mechanism for the the payment of the fees by the committee and who pays the various
+types of fees is still under discussion.
+
+#### Checkpoints
+VNs periodically post checkpoint summaries to the [base layer] for each asset that they are managing. The checkpoints will form an immutable
+record of the [Digital Asset] state on the base-layer. There will be two types of checkpoints:
+* An Opening checkpoint (OC) will:
+  * Specify the members of the VN committee.
+  * Lock up the collateral for the committee members for this checkpoint period.
+  * Collect the fee pool for this checkpoint period from the Asset Issuer into a Multi-Sig UTXO under the control of the committee.
+  This can be a top-up of the fees or a whole new fee pool.
+
+* A Closing checkpoint (CC) will:
+  * Summarize the Digital Asset state on the base layer.
+  * Release the fee pay outs.
+  * Release the collateral for the committee members for this checkpoint period.
+  * Allow for committee members to resign from the committee
+
+After an DA is issued there will immediately be an Opening checkpoint. After a checkpoint period there will then be a Closing checkpoint followed
```

Comment:
My gut says we could run into issues doing it like this (race conditions being one), but it's something that needs to be fleshed out in a sub-RFC anyway, and before this RFC hits `stable`

Reply:
Yes, I am going to leave this as is for now as the nature of the checkpoints is still a big question we need to figure out but there will be checkpoints.